source/fe/fe_system.cc

```diff
@@ -2302,6 +2303,29 @@ convert_generalized_support_point_values_to_dof_values (const std::vector<Vector
           continue;
         }
 
+      // If the base element is non-interpolatory, assign NaN to all DoFs
+      // associated to it.
+      if (!base_element.has_generalized_support_points())
+        {
+          base_dof_values.resize(n_base_dofs);
+
+          // To do this, we could really use a base_to_system_index()
+          // function, but that doesn't exist -- just do it by using the
+          // reverse table -- the amount of work done here is not worth
+          // trying to optimizing this.
+          for (unsigned int m = 0; m < multiplicity; ++m)
+            for (unsigned int i = 0; i < this->dofs_per_cell; ++i)
+              if (this->system_to_base_index(i).first ==
+                  std::make_pair(base, m))
+                dof_values[i] = std::numeric_limits<double>::quiet_NaN();
```

Comment:
I am not sure whether it is safe to use a `signaling_NaN` here. Or whether we actually want that.

Reply:
I like using a `numbers::signaling_nan<double>`. -- in essence, you are saying "*** these values are known to be invalid ***" in the return object. That makes sense to me.

But this should be documented in the function's public interface. The only complication (also worth noting in the function doc) is that one cannot test for a value to be a signaling NaN. In general, one shouldn't even try to *read* it. But one doesn't have to -- you will know which elements of the output vector to ignore in the place where you're calling the current function